lib/Option/SanitizerOptions.cpp

```diff
@@ -182,6 +182,36 @@ OptionSet<SanitizerKind> swift::parseSanitizerArgValues(
             + toStringRef(SanitizerKind::Thread)).toStringRef(b2));
   }
 
+  // Scudo can only be run with ubsan.
+  if (sanitizerSet & SanitizerKind::Scudo) {
+    OptionSet<SanitizerKind> allowedSet;
+    allowedSet |= SanitizerKind::Scudo;
+    allowedSet |= SanitizerKind::Undefined;
+
+    auto forbiddenOptions = sanitizerSet - allowedSet;
+
+    if (forbiddenOptions) {
+      SanitizerKind forbidden;
+
+      if (forbiddenOptions & SanitizerKind::Address) {
```

Comment:
Note: this if-else needs to be extended whenever the SanitizerKind enum gets extended, but I don't see a better way of doing this with the current API of `OptionSet`.

Reply:
Yup. `OptionSet` absolutely could answer this question, but I don’t know how much value we get from ramming that new single-use API into this patch.

Reply:
Not requesting anything additional for this PR, just wanted to point it out; maybe someone has an idea.

Reply:
Ah sorry, I understood you weren’t asking for new API, I was just adding clarification on why I didn’t add it as part of this patch.